styleguide/source/assets/js/webforms.js

```diff
@@ -1,110 +1,45 @@
-    input.next().remove('.form-item--error-message');
-    input.after('<div class="form-item--error-message">Field is required.</div>');
-  }
-
-  function checkField(input) {
-    if (input.prop('required') && (input.val().length === 0 || input.val() === "")) {
-      fieldIsRequired(input);
-    }
-    else {
-      if (input.attr('type') === 'email' && !validateEmail(input.val())) {
-        fieldIsRequired(input);
-      }
-      else {
-        input.removeClass('error').next().remove('.form-item--error-message');
-      }
-    }
-  }
-
-  var initialLoad = true;
-
-  Drupal.behaviors.webForm = {
+  Drupal.behaviors.formValidate = {
     attach: function (context, settings) {
-      $(document).ready(function(e) {
-        $contactForm = $('.webform-submission-form');
-        $inputs = $('.webform-submission-form section *').filter(':input');
-        $iconElement = $('.ama__form-steps__icon');
-        $submitBuutton = $('.webform-button--submit');
-
-        if (initialLoad === false) {
-          verifyFields($contactForm);
-        }
-
-        $inputs.on('focus change keypress selectmenuchange', function () {
-          var iconClass = 'edit';
-          $closestSection = $(this).closest('section');
-          $closestSectionInputs = $closestSection.find(':input');
-          $closestSection.find($iconElement).removeClass('edit error completed').addClass(iconClass);
-          $closestSectionInputs.each(function () {
-            checkField($(this));
-          });
-        });
-
-        $inputs.on('focus blur', function () {
-          var iconClass = 'edit';
-          $closestSection = $(this).closest('section');
-          $closestSectionInputs = $closestSection.find(':input');
-          $allFieldsReady = true;
-
-          $closestSectionInputs.each(function () {
-            if ($(this).prop('required') && $(this).val().length === 0) {
-              $allFieldsReady = false;
-              iconClass = 'error';
+      (function ($) {
+        // On webform submit check to see if all inputs are valid
+        $('.webform-submission-form').validate({
+          ignore: [],
+          errorPlacement: function(error, element) {
+            if (element.attr("type") == "checkbox") {
+              error.insertAfter(element.parent().siblings().last());
+            } else if (element.is("select")) {
+              error.insertAfter(element.next())
+            } else error.insertAfter(element);
+          },
+          invalidHandler: function(form, validator) {
+            var errors = validator.numberOfInvalids();
+            if (errors) {
+              $('.progress-marker').addClass('error')
             }
-          });
+          }
+        });
 
-          if ($allFieldsReady) {
-            iconClass = 'completed';
+        // Check to see if inputs are valid
+        $('.webform-submission-form input').keyup(function() {
+          if($(this).valid()) {
+            $('.progress-marker').removeClass('error')
```

Comment:
@designerneil You're missing a few semicolons here.

Reply:
https://www.drupal.org/docs/develop/standards/javascript/javascript-coding-standards#semicolons